src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,71 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * TaskList is the class that contains the list of tasks.
+ * It provides operations to add and delete tasks from the list,
+ * and mark tasks as complete.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Class constructor.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Class constructor specifying an initial list of tasks.
+     *
+     * @param tasks the list of tasks to initialise with.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        if (tasks != null) {
+            this.tasks = tasks;
+        }
+        else {
+            this.tasks = new ArrayList<>();
+        }
+    }
+
+    public int getSize() {
+        return tasks.size();
+    }
+
+    public Task getTask(int taskNum) {
```

Comment:
Perhaps it would be good to add a header comment here to tell users that taskNum is using 1-based indexing.
This applies to other methods of TaskList too.

Reply:
That's a great point, I have fixed it. Thank you!